bytecheck_derive/src/lib.rs

```diff
@@ -199,6 +199,7 @@ fn derive_check_bytes(mut input: DeriveInput) -> Result<TokenStream, Error> {
                 });
 
                 quote! {
+                    #[allow(unused_results)]
```

Comment:
I think if we use `#[automatically_derived]` then that should cover this and other lints. That might work on the `const _: () = { ... };` block that it outputs in the end. Could you try that attribute and see if that works as well?

Reply:
It looks like that only applies directly on impl blocks (if you write it outside of a macro, not on an impl, it gives a warning saying this), so I added it to all the impls directly. It doesn't seem to suppress unused_results, so I moved that to the const block.